### PR TITLE
use proxy env if available

### PR DESCRIPTION
**Pull request by hgw77**
for some reason Net::HTTP did not recognizes the http_proxy ENV variable, so I added an if case to use http_proxy if available

### CODE REVIEW

**Comment by Carthaca** [comment]
lgtm

File: lib/misty/http/net_http.rb [hunk 1]
```diff
@@ -12,4 +16,4 @@ def net_http(uri, ssl_verify_mode, log)
       end
     end
   end
-end
+end
```
Comment (gildub) [inline lib/misty/http/net_http.rb:19]: Could you please restore the newline at end of file?

**Review by gildub — CHANGES_REQUESTED** [review]
Sorry for long delay, I was on leave!
👍 for adding this.

Please review comments.

File: lib/misty/http/net_http.rb [hunk 2]
```diff
@@ -3,6 +3,10 @@ module HTTP
     module NetHTTP
       def net_http(uri, ssl_verify_mode, log)
         http = Net::HTTP.new(uri.host, uri.port)
```
Comment (gildub) [inline lib/misty/http/net_http.rb:5]: What about having this one (the non proxy http) in the else block?

Comment (hgw77) [reply lib/misty/http/net_http.rb:5]: I am not sure that makes sense because you have to declare in any case the http variable outside the if/else block

**Comment by hgw77** [comment]
I messed something up and will redo my pull request

**Pull request closed without merging**